src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs

```diff
@@ -130,11 +130,15 @@ public override bool Equals(object obj)
 
             public override int GetHashCode()
             {
-                return Hash.Combine(_descriptor,
-                        Hash.Combine(_messageArgs.GetHashCode(),
-                         Hash.Combine(_location.GetHashCode(),
-                          Hash.Combine(_severity.GetHashCode(), _warningLevel)
-                        )));
+                int hash = 0;
+
+                hash = Hash.Combine(_location.GetHashCode(), 0);
+                hash = Hash.Combine((int)_severity, hash);
```

Comment:
What about the descriptor? It could be null too. A diagnostic with no descriptor does not really make sense so should an ArgumentNullException be thrown?